orchestrator-salt/src/main/resources/salt-common/salt/filecollector/scripts/preflight-check.sh

```diff
@@ -0,0 +1,338 @@
+  run_command "$SALT_BIN_PREFIX/salt -L $preflight_check_targets cp.get_file salt:///distribution/preflight-check.sh /tmp/preflight-check.sh"
+  run_command "$SALT_BIN_PREFIX/salt -L $preflight_check_targets cmd.run 'chmod 750 /tmp/preflight-check.sh && /tmp/preflight-check.sh minion-check ${EXTRA_ARGS}' --out-indent=-1 --out=json --out-file=$WORKING_DIR/preflight_check_out.json timeout=60"
+
+  for key in "${!LOGS_TO_CHECK[@]}"; do
+    rm -rf "$WORKING_DIR/preflight_${key}_log_errors.txt"
+  done
+
+  for row in $(cat "$WORKING_DIR/preflight_check_out.json" | jq -r 'to_entries | . []|=.key+":"+.value | @base64'); do
+    local decoded_row=$(echo "${row}" | base64 -d | jq .[] | sed -r 's/^"|"$//g')
+    local salt_minion_node=$(echo "${decoded_row}" | cut -d ':' -f1)
+    local full_value=$(echo "${decoded_row}" | cut -d ':' -f2- | tr -d '\\')
+    if [[ "$full_value" == *do_not_return_start* ]]; then
+      log "Salt minion with name '${salt_minion_node}' is not responding."
+    else
+      for key in "${!LOGS_TO_CHECK[@]}"; do
+        local log_size=$(echo "$full_value" | jq ".${key}")
+        check_minion_file_size $salt_minion_node $log_size "$WORKING_DIR/preflight_${key}_log_errors.txt"
+      done
+    fi
+  done
+
+  local error_msg=""
+  for key in "${!LOGS_TO_CHECK[@]}"; do
+    local error_nodes=""
+    if [[ -f "$WORKING_DIR/preflight_${key}_log_errors.txt" ]]; then
+      error_nodes=$(cat "$WORKING_DIR/preflight_${key}_log_errors.txt" | paste -sd "," -)
+      if [[ "$error_nodes" != "" ]];then
+        error_msg=$(echo -e "$error_msg \n[${LOGS_TO_CHECK[$key]}]: $error_nodes")
+      fi
+    fi
+  done
+
+  # cleanup
+  rm -rf /srv/salt/distribution
+
+  if [[ "$error_msg" != "" ]];then
+    local full_error_msg=$(cat << EOF
+PreFlight diagnostics check FAILED
+Diagnostics log collection has been interrupted in order to avoid memory pressure on VM nodes that could shut down the machines , as some files are too large to process.
+Please upgrade cdp-telemetry binary to workaround the issue (at least to version 0.4.13) or filter out affected hosts/labals (in case of CDP cli was used for the bundle collection).
+The following logs/nodes are affected: $error_msg
+
+You can upgrade the cdp-telemetry binary by using --update-package option by CDP cli (for environment/datalake/datahub collect-XX-diagnostics commands). But note that requires network access for https://cloudera-service-delivery-cache.s3.amazonaws.com (on the VM nodes).
+Alternatively, the binary can be updated manually by the following steps (e.g. for version $EXPECTED_MIN_VERSION):
+1. Download rpm from https://cloudera-service-delivery-cache.s3.amazonaws.com/telemetry/cdp-telemetry/cdp_telemetry-$EXPECTED_MIN_VERSION.x86_64.rpm
+2. Scp the rpm file to a salt-master node. (e.g. into /home/cloudbreak folder)
+3. Execute the following commands as root on the salt-master node:
```

Comment:
when you say `salt-master` node, do you mean the master node of datalake, freeipa etc.?

Reply:
mostly, but not always true, like can be called gateway as well